contract/bridge/contracts/VaultReward.sol

```diff
@@ -0,0 +1,175 @@
+
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+import "./IVaultRegistry.sol";
+
+contract VaultReward is Initializable {
+  using SafeMathUpgradeable for uint256;
+
+  struct LockedVault {
+    uint256 lockStartAt;
+    uint256 lockPeriod;
+    uint256 lockExpireAt;
+    uint256 rewardClaimAt;
+    uint256 collateralUpdatedAt;
+    uint256 accClaimableRewards;
+  }
+
+  address public oneBtc;
+  mapping(address => LockedVault) public lockedVaults;
+
+  event ExtendVaultLockPeriod(address indexed vaultId, uint256 oldLockPeriod, uint256 newLockPeriod);
+  event ClaimRewards(address indexed vaultId, uint256 amount, uint256 claimAt);
+  event UpdateVaultAccClaimableRewards(
+    address indexed vaultId,
+    uint256 oldAccClaimableRewards,
+    uint256 newAccClaimableRewards,
+    uint256 updatedAt
+  );
+
+  modifier onlyOneBtc() {
+    require(msg.sender == oneBtc, "Only OneBtc");
+    _;
+  }
+
+  modifier vaultExist(address _vaultId) {
+    (uint256 btcPublicKeyX,,,,,,,,) = IVaultRegistry(oneBtc).getVault(_vaultId);
+    require(btcPublicKeyX != 0, "Vault does not exist");
+    _;
+  }
+
+  function initialize(address _oneBtc) public initializer {
+    oneBtc = _oneBtc;
+  }
+
+  function extendVaultLockPeriod(address _vaultId, uint256 _lockPeriod) external vaultExist(_vaultId) {
+    require(_vaultId == msg.sender, "Invalid vaultId");
+
+    // check if the lockPeriod is valid
+    require(_lockPeriod == 3 || _lockPeriod == 6 || _lockPeriod == 12, "Lock period should be one of 3, 6, 12");
+
+    // update vault accClaimableRewards
+    _updateVaultAccClaimableRewards(_vaultId);
+
+    // get vault
+    LockedVault storage vault = lockedVaults[_vaultId];
+
+    // get the current vault lock period
+    uint256 oldLockPeriod = vault.lockPeriod;
+
+    // update the vault lock info
+    uint256 secPerDay = 60 * 60 * 24;
+    uint256 lockPeriodInSec = secPerDay.mul(30).mul(_lockPeriod);
+    if (vault.lockExpireAt < block.timestamp) {   // new or expired vault
+      vault.lockStartAt = block.timestamp;
+      vault.lockPeriod = _lockPeriod;
+      vault.lockExpireAt = block.timestamp.add(lockPeriodInSec);
+      vault.rewardClaimAt = block.timestamp;
+      vault.collateralUpdatedAt = block.timestamp;
+      vault.accClaimableRewards = 0;
+    } else {    // locked vault
+      vault.lockPeriod = vault.lockPeriod.add(_lockPeriod);
+      vault.lockExpireAt = vault.lockExpireAt.add(lockPeriodInSec);
+    }
+
+    emit ExtendVaultLockPeriod(_vaultId, oldLockPeriod, vault.lockPeriod);
+  }
+
+  function _updateVaultAccClaimableRewards(address _vaultId) internal {
+    if (block.timestamp <= lockedVaults[_vaultId].lockExpireAt) {
+      // get vault
+      LockedVault storage vault = lockedVaults[_vaultId];
+
+      // store the old accClaimableRewards
+      uint256 oldAccClaimableRewards = vault.accClaimableRewards;
```

Comment:
I don't see it goes anywhere. What happens to the "old" claim rewards?

Reply:
It's for the event.
https://github.com/harmony-one/onebtc/blob/vault-incentive-implementation/contract/bridge/contracts/VaultReward.sol#L94